docs/content/how-to/troubleshooting.md

```diff
@@ -0,0 +1,90 @@
+# Troubleshooting
+
+## How to dump the HostedCluster resources from a management cluster
+
+In order to dump the relevant HostedCluster objects we will need some prerequisites:
+
+- `cluster-admin` access to the management cluster
+- The HostedCluster `name` and the `namespace` where the CR is deployed
+- The Hypershift CLI
+
+Once we have these elements and the shell with the Kubeconfig loaded pointing to the management cluster, we will execute these commands:
+
+```bash
+CLUSTERNAME="samplecluster"
+CLUSTERNS="clusters"
+
+mkdir clusterDump-${CLUSTERNS}-${CLUSTERNAME}
+hypershift dump cluster \
+    --name ${CLUSTERNAME} \
+    --namespace ${CLUSTERNS} \
+    --artifact-dir clusterDump-${CLUSTERNS}-${CLUSTERNAME}
+```
+
+After some time, the output will show something like this:
+
+```bash
+2023-06-06T12:18:20+02:00	INFO	Archiving dump	{"command": "tar", "args": ["-cvzf", "hypershift-dump.tar.gz", "cluster-scoped-resources", "event-filter.html", "namespaces", "network_logs", "timestamp"]}
+2023-06-06T12:18:21+02:00	INFO	Successfully archived dump	{"duration": "1.519376292s"}
+```
+
+This dump contains so much and quite useful information that will help on the troubleshooting procedure.
+
+### Impersonation
+
+The dump command can be used with the flag `--as`, which works in the same way as the `oc` client. If you execute the command with the flag, the CLI will impersonate all the queries against the management cluster, using that username or service account.
+
+The service account should have enough permissions to query all the objects from the namespaces, so cluster-admin is recommended to make sure you have enough permissions. The service account should be located (or have permissions to query) at least the HostedControlPlane namespace.
+
+!!! note
+
+    If your user/sa doesn't have enough permissions, the command will be executed and will dump only the objects you have permissions to get and during that process some `forbidden` errors will be raised.
+
+- Impersonation Sample using a service account:
+
+```bash
+CLUSTERNAME="samplecluster"
+CLUSTERNS="clusters"
+SA="samplesa"
+
+mkdir clusterDump-${CLUSTERNS}-${CLUSTERNAME}
+hypershift dump cluster \
+    --name ${CLUSTERNAME} \
+    --namespace ${CLUSTERNS} \
+    --as "system:serviceaccount:${CLUSTERNS}-${CLUSTERNAME}:${SA}" \
```

Comment:
Instead of ${CLUSTERNS}-${CLUSTERNAME} I would have a separate variable like `${SA_NAMESPACE}`
The service account doesn't need to live in the control plane namespace.